UpsertSubscriptionToMigrate/handler.ts

```diff
@@ -0,0 +1,268 @@
+            AND "ServicesMigration"."Services"."serviceVersion" < "excluded"."serviceVersion"
+    ` as NonEmptyString;
+
+export const log = (d: unknown): void => {
+  throw new Error(`To be implement ${d}`);
+};
+
+export const storeDocumentApimToDatabase = (
+  apimClient: ApiManagementClient,
+  config: IConfig,
+  pool: PoolClient,
+  telemetryClient: ReturnType<typeof initTelemetryClient>
+) => (
+  retrievedDocument: RetrievedService
+): TE.TaskEither<DomainError, QueryResult | void> =>
+  pipe(
+    retrievedDocument.serviceId,
+    // given the subscription, retrieve it's apim object
+    id => getApimOwnerBySubscriptionId(config, apimClient, id),
+    TE.chain(apimSubscription =>
+      pipe(
+        // given the subscription apim object, retrieve its owner's detail
+        getApimUserBySubscription(config, apimClient, apimSubscription),
+        // We only consider subscription owned by a Delegate,
+        //   otherwise we just ignore the document
+        // This because migration are meant to work only from a Delegate to its Organization,
+        //   not to migrate subscriptions between organizations
+        TE.chain(apimUser =>
+          ApimDelegateUserResponse.is(apimUser)
+            ? // continue processing incoming document
+              pipe(
+                { apimSubscription, apimUser },
+                apimData => mapDataToTableRow(retrievedDocument, apimData),
+                createUpsertSql(config),
+                sql => queryDataTable(pool, sql),
+                res => {
+                  trackProcessedServiceDocument(telemetryClient)(
+                    retrievedDocument
+                  );
+                  return res;
+                }
+              )
+            : // processing is successful, just ignore the document
+              TE.of<DomainError, QueryResult | void>(
+                trackIgnoredIncomingDocument(telemetryClient)(retrievedDocument)
+              )
+        )
+      )
+    )
+  );
+
+export const createHandler = (
+  config: IConfig,
+  apimClient: ApiManagementClient,
+  client: Pool,
+  telemetryClient: ReturnType<typeof initTelemetryClient>
+): Parameters<typeof withJsonInput>[0] =>
+  withJsonInput(
+    async (context, item): Promise<void> => {
+      const logPrefix = context.executionContext.functionName;
+      const pool = await client.connect();
+      return pipe(
+        item,
+        IncomingQueueItem.decode,
+        E.mapLeft(failures => {
+          const err = new Error(readableReport(failures));
+          context.log(`${logPrefix}|Invalid incoming message: ${err.message}`);
+          return err;
+        }),
+        TE.fromEither,
+        TE.chainW(
+          flow(
+            _ => _.service,
+            storeDocumentApimToDatabase(
+              apimClient,
+              config,
+              pool,
+              telemetryClient
+            ),
+            TE.mapLeft(err => {
```

Comment:
ditto

Reply:
done in https://github.com/pagopa/io-subscription-migration/pull/15/commits/fee5f6a47347aa0577d0968e988835533ca3a108